_posts/2019-02-14-Viewability.md

```diff
@@ -0,0 +1,115 @@
+Therefore, we use machine learning as approach. The general idea is to weight each individual bid by the probability of viewability.
+
+#### Training
+
+We train two independent machine learning models, one for the measurability and one for the viewability. The measurement is a mainly technical problem and therefore is influenced by the browser version, the publisher, or the operating system.
+The viewability is influenced mostly by the publisher, the adunit itself, and the position on the site. Since the relation between those features are very different for the different metrics, we separate them in independent models.
+Based on historic data, we have a ground truth dataset in the form:
+> impression ID | measurable | viewable | features
+> 1234 | 1 | 0 | *list of features*
+> 1235 | 1 | 1 | *list of features*
+
+The features consist of various properties of the context (e.g. app/site, position on the site), the content (e.g. size of the adunit, and the user and device (e.g. operating system, browser version)
+For the measurement model we take all impressions and use the measurable column as target in a supervised classification model (namely a Neural Network).
+The output of this model is the probability of each impression to be measurable. The viewability model takes all impressions, which are measurable and uses the viewability column as target.
+This model gives a conditional probability of viewability for impressions which are measurable. Thus, the total probability of viewability is:
+
+TODO: Insert nice looking formula here
+
+We perform a training of this model every couple of hours to be sure to make use of all recent campaign and publisher data.
+
+#### Application
+
+We can use the combination of the two models in our real-time bidding platform to get the probability for each impression we want to bid on. Then, we can scale the bidding price with this probability.
+So, if a viewable impression has the value of one cent to us and the probability of a view is 50%, we are bidding half a cent. If we buy two impressions with that probability for that price, we will get one viewable impression on average and pay one cent.
+Mission accomplished. There are two interesting aspects to this: First, we do not buy the viewable impressions for the cheapest possible price. This is approached with another mechanism. We are buying the viewable impression for a defined price and can allways be sure, to buy for that price or lower.
+This allows us to sell the viewable impression for a specific price. If we configure a price, which is too low we might not be able to buy the desired amount of impressions, but we will not over-pay.
```

Comment:
suggest "This allows us to offer viewable impression for a defined price to our customers."